subprojects/docs/src/docs/userguide/api/kotlin_dsl.adoc

```diff
@@ -702,6 +702,55 @@ tasks.create("myTask") {
 // === `Property`, `Provider` and `NamedDomainObjectProvider`
 
 
+[[kotdsl:assignment]]
+== Kotlin Property assignment
+
+Gradle's Kotlin DSL supports Property assignment as an experimental feature that can be enabled by adding `systemProp.org.gradle.unsafe.kotlin.assignment=true` to the `gradle.properties` file.
+Property assignment reduces the verbosity for Kotlin DSL when <<lazy_configuration#lazy_properties,lazy properties>> are used.
+Property assignment works for properties that are _final_ and have type _Property_ or _ConfigurableFileCollection_.
+Property assignment should be preferred over custom property setters since it makes Kotlin DSL code style consistent.
+
+.Kotlin Property assignment
+====
+include::sample[dir="snippets/kotlinDsl/assignment/kotlin",files="build.gradle.kts[tags=assignment]"]
+====
+<1> Set value old way, with `.set()` method (still supported)
+<2> Set value with Property assignment with `=` operator
+<3> Property assignment also supports lazy values
+
+[CAUTION]
+====
+Enabling Kotlin Property assignment will fail your build if `allWarningsAsErrors` feature is enabled for the Kotlin compiler for build logic, since the Kotlin compiler plugin is still experimental.
```

Comment:
:x: Let's link to the `allWarningsAsErrors` feature here.

Reply:
Ah, sorry, I thought this was ours. Another option is to qualify that this is a Kotlin option. My original intent was to add more context, so I should have said that instead of proposing to link to docs (which is just one way of adding context).

Reply:
Removed the link in the end.